src/UXClient/Components/GeoProcessGraphic/GeoProcessGraphic.ts

```diff
@@ -0,0 +1,170 @@
+    graphicSrc: string,
+    data: Array<TsqExpression>,
+    chartOptions
+    ) {
+    this.zoom = chartOptions.zoom;
+    this.subscriptionKey = chartOptions.subscriptionKey;
+    this.width = chartOptions.width;
+    this.height = chartOptions.height;
+    this.img = graphicSrc;
+    this.theme= chartOptions.theme;
+    this.renderBase(environmentFqdn, getToken, graphicSrc, data, chartOptions);
+  }
+
+  protected loadGraphic(graphicSrc: string): Promise<GraphicInfo> {
+    (<HTMLElement>this.component.node()).style.width = `${this.width}px`;
+    (<HTMLElement>this.component.node()).style.height = `${this.height}px`;
+    
+    let map = new atlas.Map(<HTMLElement>this.component.node(), {
+      authOptions: {
+        authType: atlas.AuthenticationType.subscriptionKey,
+        subscriptionKey: this.subscriptionKey
+      }
+    });
+    map.events.add('ready', () => {
+        this.dataSource = new atlas.source.DataSource();
+        map.sources.add(this.dataSource);
+        this.popup = new atlas.Popup({
+            content: `<div style="padding:10px;"></div>`,
+            pixelOffset: [0, -30]
+        })
+        this.marker =  new atlas.HtmlMarker({
+            htmlContent: '<img class="circleImage " src= "' + this.img + '" />',
+            position: [0, 0],
+            popup: this.popup
+        });
+        map.markers.add(this.marker);
+        map.events.add('click', this.marker, () => {
+            this.marker.togglePopup();
+        });
+    });
+    map.setCamera({
+        center: [0, 0],
+        bearing: 0,
+        pitch: 90,
+        zoom: this.zoom,
+        maxZoom : 4,
+        minZoom: 1.5,
+        type: "fly",
+        duration: 100
+    });
+    return Promise.resolve({
+        graphic: map,
+        width: this.width,
+        height: this.height
+      });
+  }
+  
+  protected createDataPoints(prm: Array<IGeoProcessGraphicLabelInfo>){
+    let dataPoints = prm.map((r, i): IGeoProcessGraphicLabelInfo => {
+        let result = this.parseTsqResponse(r);
+        this.positionXVariableName = this.tsqExpressions[i].positionXVariableName;
+        this.positionYVariableName = this.tsqExpressions[i].positionYVariableName;
+        this.alias = this.tsqExpressions[i].alias;
+        return result;  
+      });
+      this.updateDataMarkers(dataPoints);
+  }
+
+  protected parseTsqResponse(response) {
+      let parsedResponse = {};
+    if (response && response.properties){
+        for (let i =0; i<response.properties.length; i++){
+            response.properties[i] && response.properties[i].name && response.properties[i].values ?
+                parsedResponse[response.properties[i].name] = response.properties[i].values[0]
+                : null;
+        }
+    }
+    return parsedResponse;
+  }
+
+  protected updateDataMarkers(dataPoints: Array<any>) {
+        for(let i = 0; i < dataPoints.length; i++) {
+            let lat = dataPoints[i][this.positionXVariableName];
```

Comment:
yeah as i said above, this is particular to a tsqExpression, not a GeoProcessGraphic